extend-string.js

```diff
@@ -0,0 +1,114 @@
+/* Checks if a string contains vowels */
+String.prototype.hasVowels = function () {
+  return (/[aeiou]/g).test(this); //use .test to match regEx patter against string to return boolean
+};
+
+/*Converts entire string to uppercase letter*/
+String.prototype.toUpper = function () {
+  let capitalizedString = ""; //empty string variable to store result
+  for(let index = 0; index < this.length; index++) { //loop through the string to get each letter
+    let characterCode = this.charCodeAt(index); //get the character code of each character
+    if(characterCode >= 97 && characterCode <= 122) { //check if character code is in lowercase using ascii code
+      characterCode -= 32; //subtract 32 from character code to get uppercase equivalent character code
+    }
+    capitalizedString += String.fromCharCode(characterCode); //get letter from character code
+  }
+  return capitalizedString;
+};
+
+/*Converts entire string to lowercase letter*/
+String.prototype.toLower = function () {
+  let lowercaseString = ""; //empty string variable to store result
+  for(let index = 0; index < this.length; index++) { //loop through the string to get each letter
+    let characterCode = this.charCodeAt(index); //get the character code of each character
+    if(characterCode >= 65 && characterCode <= 90) { //check if character code is in uppercase using ascii code
+      characterCode += 32; //add 32 from character code to get lowercase equivalent character code
+    }
+    lowercaseString += String.fromCharCode(characterCode); //get character equivalent from character code
+  }
+  return lowercaseString;
+};
+
+/*Changes the first letter of a string to uppercase*/
+String.prototype.ucFirst = function () {
+  if(/^[a-z]/.test(this)) { //check if string starts with a small letter
```

Comment:
It is advisable if you check for the negative condition first. That way you can remove the braces around the if statement and achieve zero-level indentation.